File: Stack/daily-temperatures/daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temperatures) {
    const res = new Array(temperatures.length).fill(0);
    const stack = [];

    for (let i = 0; i < temperatures.length; i++) {
        while (
            stack.length > 0 &&
            temperatures[i] > temperatures[stack.at(-1)]
        ) {
            const stackInd = stack.pop();
            res[stackInd] = i - stackInd;
        }

        stack.push(i);
    }

    return res;
};

File: Stack/daily-temperatures/daily-temperatures.cs
public class Solution
{
    public int[] DailyTemperatures(int[] temperatures)
    {
        int[] res = new int[temperatures.Length];
        Stack<int> stack = new Stack<int>();

        for (int i = 0; i < temperatures.Length; i++)
        {
            while (stack.Count > 0 && temperatures[i] > temperatures[stack.Peek()])
            {
                int stackInd = stack.Pop();
                res[stackInd] = i - stackInd;
            }

            stack.Push(i);
        }

        return res;
    }
}